client/client.go

```diff
@@ -194,6 +201,9 @@ func (client *Client) raw(method, urlpath string, query url.Values, headers map[
 	if err != nil {
 		return nil, RequestError{err}
 	}
+	if client.userAgent != "" {
+		req.Header.Set("User-Agent", client.userAgent)
```

Comment:
we have also httputil.SetUserAgentFromVersion  which is used for interaction with the store, we need to step back a bit or this will get confusing overall

Reply:
Very good point, I updated the code in cmd/snap/main.go to use the httputil.UserAgent() for the default value when using the `snap` command.